### PR TITLE
Add example for `request` + "Garbage Collection"

### PR DESCRIPTION
**Pull request by WTCT-TOP**
## This relates to...

See #3914

## Changes

Added an example for `request` + "Garbage Collection".

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [x] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -294,6 +294,22 @@ const headers = await fetch(url)
   .then(res => res.headers)
 ```
 
+The same applies for `request` too:
+```js
+// Do
+const response = await request(url)
+  .then(res => {
+    response.body.dump().catch(err => {
+      console.error('Error while dumping the body');
```
Comment (WTCT-TOP) [inline README.md:303]: Is there anything to do here? Will the body be cleared in this case?

Comment (ronag) [reply README.md:303]: dump can't throw

**Comment by KhafraDev** [comment]
It was added under the fetch specification compliance section. It should be moved elsewhere.

**Review by mcollina — COMMENTED** [review]
Can you remove the unnecessary promises?

**Comment by WTCT-TOP** [comment]
@KhafraDev do you think I should move the whole GC section or just the new example for request?
@mcollina what do you mean by "unnecessary promises"? use `async/await` instead of `.then/.catch`? I followed the existing example..

Also, is there a reason to `await res.body.dump()` if I don't care about the body? isn't `res.body.dump()` (without await) good enough?

**Comment by mcollina** [comment]
> @mcollina what do you mean by "unnecessary promises"? use async/await instead of .then/.catch? 

yes exactly. Currently this allocates one promise too much ;).

> Also, is there a reason to await res.body.dump() if I don't care about the body? isn't res.body.dump() (without await) good enough?

If you don't await dump, you'll still have asynchronous activity happening in the background, as the body will not be dumped immediately. Not awaiting it defeats the purpose of the example, as without it there will be lingering memory and activity running.

**Comment by WTCT-TOP** [comment]
I've updated both examples to use async await, let me know what you think..

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**